torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -517,6 +511,20 @@ ExecutionPlan ProfilingGraphExecutorImpl::getPlanFor(
   return *optimized_plan_;
 }
 
+const ExecutionPlan& ProfilingGraphExecutorImpl::getPlanFor(
+    Stack& stack,
+    size_t remaining_bailout_depth) {
+  std::lock_guard<std::mutex> lock(compile_mutex);
+
+  // IMPORTANT: This is a hot path of calling a torchscript function. Try not to
+  // add any code above this.
+  if (optimized_plan_) {
```

Comment:
I think we still need to check `getGraphExecutorOptimize`

Reply:
maybe in debug mode we would also update the optimized_plan_

Reply:
Yeah, good catch, I'll revert that part. It's a bit of a bummer since `getGraphExecutorOptimize` pokes at TLS and probably was partially responsible for the slowdown, but I agree that even if we're going to change that somehow, it shouldn't be a part of this PR.